plugins/utils/fzf-lua.nix

```diff
@@ -0,0 +1,89 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.fzf-lua;
+in {
+  options.plugins.fzf-lua =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "fzf-lua";
+
+      package = helpers.mkPackageOption "fzf-lua" pkgs.vimPlugins.fzf-lua;
+
+      profile = mkOption {
+        type = with types; (enum ["default" "fzf-native" "fzf-tmux" "fzf-vim" "max-perf" "telescope" "skim"]);
+        default = "default";
+        description = "Preconfigured profile to use";
+      };
```

Comment:
This should be made optional (i.e. `helpers.defaultNullOpts.mkEnumFirstDefault`).
We will then only call the `setup` if the user has explicitly provided a profile and/or options.